### PR TITLE
Set pagination result value

### LINKED ISSUE #226: Validator query returns only 10 elements
On testnet-3 `tgrade q poe validators`
* Support `--limit `

### PR DESCRIPTION
**Pull request by alpe**
Resolves #226  
- Server side: page key in result set
- CLI: page key encoding

### CODE REVIEW

File: x/poe/contract/contractio.go [hunk 1]
```diff
@@ -176,6 +180,55 @@ func (a ContractAdapter) doExecute(ctx sdk.Context, msg interface{}, sender sdk.
 	return sdkerrors.Wrap(err, "execute")
 }
 
+// RawPageableResult is a query response type were the pagination key is the raw last element.
+type RawPageableResult interface {
```
Comment (maurolacy) [inline x/poe/contract/contractio.go:184]: This interface is currently unused, for what I've seen. Do you think it makes sense to introduce it now?

File: x/poe/client/cli/query.go [hunk 2]
```diff
@@ -392,3 +394,20 @@ func AddPaginationFlagsToCmd(cmd *cobra.Command, query string) {
 	cmd.Flags().String(flags.FlagPageKey, "", fmt.Sprintf("pagination page-key of %s to query for", query))
 	cmd.Flags().Uint64(flags.FlagLimit, 100, fmt.Sprintf("pagination limit of %s to query for", query))
 }
+
+// sdk ReadPageRequest expects binary but we encoded to base64 in our marshaller
+func withPageKeyDecoded(flagSet *flag.FlagSet) *flag.FlagSet {
```
Comment (maurolacy) [inline x/poe/client/cli/query.go:399]: This was the real issue I guess?

The rest looks nice, but it's in the end generalization and encapsulation of the pagination key.

File: x/poe/contract/contractio.go [hunk 3]
```diff
@@ -176,6 +180,55 @@ func (a ContractAdapter) doExecute(ctx sdk.Context, msg interface{}, sender sdk.
...
+
+	if err := json.Unmarshal(res, result); err != nil {
+		return nil, sdkerrors.Wrap(err, "unmarshal result")
+	}
+
+	var cursor PaginationCursor
+	switch p := result.(type) {
+	case PageableResult:
+		cursor = p.PaginationCursor()
+	case RawPageableResult:
```
Comment (maurolacy) [inline x/poe/contract/contractio.go:217]: Same here. This is cool, but if unused, why add it now?

**Review by maurolacy — APPROVED** [review]
Looks good. Just some comments on syntax and currently unused stuff.

File: x/poe/contract/contractio.go (hunk 3 above)
Comment (alpe) [reply x/poe/contract/contractio.go:217]: I appreciate the feedback and you have an 👁️  on not over engineering the query! 
I had started with the "raw" query type as this was the common pagination way as far as I remembered. But this is not correct for this contract type.  I will remove this now.

**Pull request merged**